### PR TITLE
Trajectory interface

### PR DESCRIPTION
**Pull request by fmauch**
Adds the trajectory interface from the [beta-testing](https://github.com/UniversalRobots/Universal_Robots_Client_Library/tree/beta-testing) branch.

### CODE REVIEW

**Comment by fmauch** [comment]
Setting this back to draft, as @stefanscherzinger and I plan another iteration before merging. @t-schnell it would be nice if you could review that state nevertheless, as it reflects the things that you have done before.

**Comment by stefanscherzinger** [comment]
@fmauch , I only skimmed over 5a66723 , but [UniversalRobots/Universal_Robots_ROS_Driver@01a1cf0](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/commit/01a1cf044dedec3a5c9cccf5ef583115c5e40aa5) looks good as a basis. I'll see how to get that back into the PassThroughControllers and where/how to best convert the result flags.

File: include/ur_client_library/control/trajectory_point_interface.h
Comment (t-schnell) [inline include/ur_client_library/control/trajectory_point_interface.h:41]:
```suggestion
 * \brief Types for encoding trajectory execution result.
```

File: include/ur_client_library/control/trajectory_point_interface.h
Comment (t-schnell) [inline include/ur_client_library/control/trajectory_point_interface.h:79]:
```suggestion
   * \param time The goal time to reach the target
   * \param blend_radius The radius to be used for blending between control points
   * \param cartesian True, if the written point is specified in cartesian space, false if in joint space
```

Unless thereis a reason for these params not to be documented here.

File: include/ur_client_library/ur/ur_driver.h
Comment (t-schnell) [inline include/ur_client_library/ur/ur_driver.h:189]:
```suggestion
   * \param cartesian True, if the point sent is cartesian, false if joint-based
   * \param goal_time Time for the robot to reach this point
   * \param blend_radius  The radius to be used for blending between control points
```
Included reordering to fit order of parameters in function definition.

File: include/ur_client_library/ur/ur_driver.h
Comment (t-schnell) [inline include/ur_client_library/ur/ur_driver.h:200]:
```suggestion
   * \param trajectory_action The action to be taken, such as starting a new trajectory
   * \param point_number The number of points of a new trajectory to be sent
```

Consistency with other param documentations

File: include/ur_client_library/ur/ur_driver.h
Comment (t-schnell) [inline include/ur_client_library/ur/ur_driver.h:296]:
```suggestion
   * One mode of robot control is to forward a complete trajectory to the robot for execution.
   * When the execution is done, the callback function registered here will be triggered.
```

File: include/ur_client_library/ur/ur_driver.h
Comment (t-schnell) [inline include/ur_client_library/ur/ur_driver.h:299]:
```suggestion
   * a trajectory execution
```
Consistency with other param documentations

File: resources/external_control.urscript
Comment (t-schnell) [inline resources/external_control.urscript:133]: Should we leave all of these in? In general might be worth thinking about reducing text output, but should not be critical.

File: resources/external_control.urscript
Comment (t-schnell) [inline resources/external_control.urscript:154]:
```suggestion
```
If we decide no additional explicit error handling is necessary here, this can be removed.

File: resources/external_control.urscript
Comment (t-schnell) [inline resources/external_control.urscript:173]: With the additional trajectory socket leaving beta, it might be worth considering if this is solution for setting the port is sufficient, or if we should add a way to explicitly set it via the setup.

File: src/control/trajectory_point_interface.cpp
Comment (t-schnell) [inline src/control/trajectory_point_interface.cpp:98]:
```suggestion
                   "one connection is allowed at a time. Ignoring this request.");
```
Looks weird, assume this should be just as fine.

File: src/control/trajectory_point_interface.cpp
Comment (t-schnell) [inline src/control/trajectory_point_interface.cpp:121]:
```suggestion
      URCL_LOG_DEBUG("Trajectory execution finished with result %d, but no callback was given.");
```

File: src/ur/ur_driver.cpp
Comment (t-schnell) [inline src/ur/ur_driver.cpp:157]:
```suggestion
```
As is in the control script - could be left like this for now, but worth thinking about deciding if we want to change this now.

**Review by t-schnell — CHANGES_REQUESTED** [review]
Mostly minor changes in documentation, biggest open question is setting the port for the additional socket.

File: resources/external_control.urscript
Comment (fmauch) [reply resources/external_control.urscript:133]: good point, I'll clean this up a bit.

File: resources/external_control.urscript
Comment (fmauch) [reply resources/external_control.urscript:173]: Almost forgot about that... Yes, I do not like this at all. I also don't like introducing yet another parameter, but I guess it's necessary.

Comment (fmauch) [reply resources/external_control.urscript]: Added another parameter in ca8fc29

File: src/ur/ur_driver.cpp
Comment (fmauch) [reply src/ur/ur_driver.cpp]: See ca8fc29

File: resources/external_control.urscript
Comment (fmauch) [reply resources/external_control.urscript]: Removed in 7d8236d

File: resources/external_control.urscript
Comment (fmauch) [reply resources/external_control.urscript]: Removed in 7d8236d

**Comment by fmauch** [comment]
@t-schnell I worked in all your suggestions. As you stated that otherwise this PR is alright, I will merge it on Monday if I don't hear back from you.

**Comment by t-schnell** [comment]
Looked over it (quickly), everything seems fine and can be merged.

**Pull request merged**
